[Backport] VectorAPI: 8236443: Issues with specializing vector register type for phi operand with generic operands

yyang52 opened the pull request:
[Backport] VectorAPI: 8236443: Issues with specializing vector register type for phi operand with generic operands

Summary: Fix special handling for operand resolution of vectorshift and shiftcount nodes. Fix for crash in the resolution algorithm due to non-machine type nodes.

Test Plan: ci jtreg

Reviewed-by: JoshuaZhuwj

Issue: https://github.com/alibaba/dragonwell11/issues/302

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/dragonwell11?pullRequest=303) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/dragonwell11?pullRequest=303) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/dragonwell11?pullRequest=303) it.</sub>

yyang52 commented:
@JoshuaZhuwj please help to take a review, thanks

JoshuaZhuwj reviewed (approved):
The change looks good.
However "summary" in commit message seems redundant with commit title.
It would be better describe the change in more details.
Like "Fix special handling for operand resolution of vectorshift and shiftcount nodes. Fix for crash in the resolution algorithm due to non-machine type nodes." for this case.

yyang52 commented:
> The change looks good. However "summary" in commit message seems redundant with commit title. It would be better describe the change in more details. Like "Fix special handling for operand resolution of vectorshift and shiftcount nodes. Fix for crash in the resolution algorithm due to non-machine type nodes." for this case.

Thanks, updated.

JoshuaZhuwj commented:
> > The change looks good. However "summary" in commit message seems redundant with commit title. It would be better describe the change in more details. Like "Fix special handling for operand resolution of vectorshift and shiftcount nodes. Fix for crash in the resolution algorithm due to non-machine type nodes." for this case.
> 
> Thanks, updated.

I think you have to do "git commit --amend" and then "git push xxx HEAD:JDK-8236443" so that a force push is able to update "commit message".

yyang52 commented:
> > > The change looks good. However "summary" in commit message seems redundant with commit title. It would be better describe the change in more details. Like "Fix special handling for operand resolution of vectorshift and shiftcount nodes. Fix for crash in the resolution algorithm due to non-machine type nodes." for this case.
> > 
> > 
> > Thanks, updated.
> 
> I think you have to do "git commit --amend" and then "git push xxx HEAD:JDK-8236443" so that a force push is able to update "commit message".

Updated, thanks
